File: LeetCode/49. Group Anagrams/Solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        for val in strs:
            key = ''.join(sorted(val))
            if(key in dic): dic[key].append(val)
            else: dic[key] = [val]
        result = []
        for key in dic:
            result.append(dic[key])
        return result

File: LeetCode/49. Group Anagrams/Solution.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  let dict = {};

  for (let i = 0; i < strs.length; i++) {
    let key = strs[i].split("").sort().join("");
    if (dict[key] === undefined) {
      dict[key] = [];
    }
    dict[key].push(strs[i]);
  }

  var result = [];
  for (let key in dict) result.push(dict[key]);

  return result;
};
